sacn/messages/data_packet.py

```diff
@@ -32,7 +35,18 @@ def __init__(self, cid: tuple, sourceName: str, universe: int, dmxData: tuple =
 
     def __str__(self):
         return f'sACN DataPacket: Universe: {self._universe}, Priority: {self._priority}, Sequence: {self._sequence} ' \
-               f'CID: {self.cid}'
+               f'CID: {self._cid}'
+
+    @property
+    def cid(self) -> tuple:
+        return self._cid
+
+    @cid.setter
+    def cid(self, cid: tuple):
+        if (len(cid) != 16 or not all(isinstance(x, int) for x in cid) or not all(0 <= x <= 16 for x in cid)):
+            raise TypeError(f'cid must be a 16 byte tuple! value was {cid}')
+        super().__init__(126 + len(self._dmxData), cid, VECTOR_ROOT_E131_DATA)
+        self._cid = cid
```

Comment:
The parent class `RootLayer` actually stores the CID (see `_cid`), so this property should be set there and not implemented in an inheriting class.

I'm curious: what does the line `not all(0 <= x <= 16 for x in cid)` check against? The CID is allowed to be 16 bytes and therefore each byte is allowed to be in the range of [0-255]. Doesn't this condition actually fail on valid data like `(255, 127, 0, 1, 5, 6, 7, 8, 9, 10, 11, 12, 13 ,14 , 15, 16)`?

Reply:
for comment 1 -- fair enough.  I'll move this one down to root_layer.
for comment 2 -- yes, yes it does.  Typing too quickly.  I'll fix that up